src/test/java/uk/gov/pay/api/exception/mapper/CreateChargeExceptionMapperTest.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.pay.api.exception.mapper;
+
+import org.junit.Before;
+import org.junit.Test;
+import uk.gov.pay.api.exception.ConnectorResponseErrorException.ConnectorErrorResponse;
+import uk.gov.pay.api.exception.CreateChargeException;
+import uk.gov.pay.api.model.PaymentError;
+import uk.gov.pay.commons.model.ErrorIdentifier;
+
+import javax.ws.rs.core.Response;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.pay.api.model.PaymentError.Code.CREATE_PAYMENT_MANDATE_STATE_INVALID;
+import static uk.gov.pay.api.model.PaymentError.aPaymentError;
+
+public class CreateChargeExceptionMapperTest {
+
+    private CreateChargeExceptionMapper mapper;
+    
+    @Before
+    public void setUp() {
+        mapper = new CreateChargeExceptionMapper();
+    }
+    
+    @Test
+    public void shouldThrow409_whenMandateStateInvalid(){
+        Response response = mock(Response.class);
+        when(response.readEntity(ConnectorErrorResponse.class))
+                .thenReturn(new ConnectorErrorResponse(
+                        ErrorIdentifier.MANDATE_STATE_INVALID, null, null));
+        
+        Response returnedResponse = mapper.toResponse(new CreateChargeException(response));
+        PaymentError returnedError = (PaymentError) returnedResponse.getEntity();
+        assertThat(returnedResponse.getStatus(), is(409));
```

Comment:
It’s always nice to have a blank line before the assertions so it’s clear what’s actually being tested. Personally, I like to space things so there’s the stubbing, then a blank line, then the actual call that’s tested, then a blank line, then the assertions and verifications. Like this:

```java
@Test
public void myAmazingTest() {
    when(mockSomething.getSomething()).thenReturn(foo);
    when(mockSomethingElse.getSomethingElse()).thenReturn(bar);

    String result = thisIsTheActualThingWeAreTesting.allOnItsOwnSoItsClear(whatExactlyIsBeingTested);

    assertThat(result, is(whatWeExpect));
    assertThat(result, is(awesome));

    verify(test.isEasyToRead());
    verify(and.weAre(done));
}
```